plugin/runtime/communication.py

```
@@ -140,7 +182,62 @@ def _cleanup_pending_futures(self) -> None:
+                    future = self._pending_futures.get(req_id)
+                    if future and future.done():
+                        self.logger.debug(
+                            f"Cleaning up completed Future for req_id={req_id} after timeout"
+                        )
+                    self._pending_futures.pop(req_id, None)
+            
+            cleanup_task = asyncio.create_task(cleanup_after_delay())
+            self._background_tasks.add(cleanup_task)
+            cleanup_task.add_done_callback(self._background_tasks.discard)
+            
+            raise TimeoutError(f"%s execution timed out after %ss" % (error_context, timeout)) from None
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)